### ISSUE TITLE
"autoPlay is triggered more than once" since v0.22.0

### ISSUE THREAD

**Issue by krisbulman** [akiran/react-slick#1150]
Since `0.22.0` (and latest 0.23.1), I am now getting `autoPlay is triggered more than once` repeatedly in the console. React `15.5.4`.

![react-slick](https://user-images.githubusercontent.com/224268/37970811-c824c744-31aa-11e8-9d21-9e07677fcc75.png)

```
  static defaultSliderProps = {
    arrows: true,
    autoplay: true,
    autoplaySpeed: 4000,
    dots: false,
    draggable: false,
    fade: true,
    infinite: true,
    lazyLoad: 'progressive',
    slidesToShow: 1,
    slidesToScroll: 1,
    speed: 1000,
    waitForAnimate: true,
  };
```

Reverting to `0.21.0` doesn't show this error.

Will try to replicate later on today in codesandbox.

**Comment by misterbridge** [comment]
Same issue here.
Due to lazyloading: image.onload triggers an update.
But the worst part is that is doesnt work: not yet loaded image do not pause the autoplay, so with a very bad connection, we might end up with scrolling slider of blank images.

**Comment by laveesingh** [comment]
It will be considered soon...Thanks.

**Comment by misterbridge** [comment]
Thanks @laveesingh .
Looking at your commit fe75098, I'd want to emphasis the sad part I mentionned, which is the rootcause of this issue: The lazy loading image.onload timer does'nt really works as it does not pause the autoplay. Hence a new play without a previous pause.

I had to do something like that in my implementation:
```javascript
[...]
   afterChange: this.waitForImage,
[...]

  waitForImage = index => {
    if (!this.loadedImages.includes(index)) {
      this.slider.slickPause();
    }
  };

  loaded = index => {
    if (!this.loadedImages.includes(index)) {
      this.loadedImages.push(index);
      this.sliderPlay();
    }
  };
  
[...]
  {images.map((image, idx) => (
[...]
    <img [...]
      onLoad={() => {
        this.loaded(idx);
      }}
    />
[...]
  ))}
[...]
```

Would the final fix consider that too ?

**Comment by laveesingh** [comment]
I'll surely take a look at this.

**Comment by laveesingh** [comment]
It's been fixed and will be released with 0.23.0.

**Comment by krisbulman** [comment]
cc @fender
